internals/overlord/state/notices_test.go

```diff
@@ -108,16 +112,55 @@ func (s *noticesSuite) TestUnmarshal(c *C) {
 	})
 }
 
+func (s *noticesSuite) TestString(c *C) {
+	noticeJSON := []byte(`{
+		"id": "1",
+		"user-id": 1000,
+		"type": "custom",
+		"key": "foo.com/bar",
+		"first-occurred": "2023-09-01T05:23:01Z",
+		"last-occurred": "2023-09-01T07:23:02Z",
+		"last-repeated": "2023-09-01T06:23:03.123456789Z",
+		"occurrences": 2,
+		"last-data": {"k": "v"},
+		"repeat-after": "60m",
+		"expire-after": "168h0m0s"
```

Comment:
These last three fields are optional (and `omitempty`) and not interesting for this test, so let's leave them off here and below in `TestString`.